pkg/controller/migmigration/task.go

```diff
@@ -1052,12 +1055,32 @@ func (t *Task) init() error {
 
 }
 
-func (t *Task) initPipeline() {
-	for _, phase := range t.Itinerary.Phases {
-		t.Owner.Status.AddStep(&migapi.Step{
-			Name:    phase.Step,
-			Message: "Not started",
-		})
+func (t *Task) initPipeline(prevItinerary string) error {
+	if t.Itinerary.Name != prevItinerary {
```

Comment:
For anyone looking: any time we switch itineraries, the list of steps will be pre-constructed from the list of phases.